src/excited_workflow/carbon_tracker.py

```diff
@@ -0,0 +1,275 @@
+        y_key: target variable name.
+
+    Returns:
+        Dataset for training and prediction dataset.
+    """
+    mask = df["group"] != number
+    df_train = df[mask]
+    pycs, model = train_model(df_train, x_keys, y_key)
+
+    df_prediction = df[~mask]
+    data = df_prediction[x_keys + [y_key]]
+    ds_target = xr.Dataset.from_dataframe(data)
+    data.drop(y_key, axis=1, inplace=True)
+
+    prediction = pycs.predict_model(model, data=data)
+    ds_prediction = xr.Dataset.from_dataframe(prediction)
+
+    return ds_target, ds_prediction
+
+
+def calculate_rmse(prediction: xr.DataArray, target: xr.DataArray) -> xr.DataArray:
+    """Calculate RMSE.
+
+    Args:
+        prediction: column of prediction.
+        target: column of target.
+
+    Returns:
+        RMSE
+    """
+    rmse = np.sqrt(((prediction - target) ** 2).mean(dim="time", skipna=True))
+    rmse_da = xr.DataArray(rmse)
+    return rmse_da
+
+
+def create_rmseplot(rmse: xr.DataArray) -> None:
+    """Create map plot for rmse.
+
+    Args:
+        rmse: rmse dataarray
+    """
+    plt.figure(figsize=(5, 3))
+    rmse.plot()
+    plt.tight_layout()
+
+
+def create_scatterplot(prediction: xr.DataArray, target: xr.DataArray) -> None:
+    """Create scatterplot of prediction vs. target.
+
+    Args:
+        prediction: dataframe column of prediction.
+        target: dataframe column of target variable.
+    """
+    plt.figure(figsize=(5, 5))
+    plt.scatter(prediction, target, s=25)
+    plt.xlabel("Prediction")
+    plt.ylabel("Target")
+    plt.tight_layout()
+
+
+def validate_model(
+    ds: xr.Dataset, groups: int, x_keys: list[str], y_key: str, output_dir: Path
+) -> None:
+    """Validate the trained model by calculating rmse and scatterplots.
+
+    Args:
+        ds: dataset for training.
+        groups: number of groups.
+        x_keys: list of input variables.
+        y_key: target variable name.
+        output_dir: directory to output rmse and scatterplots.
+    """
+    df_group = create_groups(ds, groups)
+
+    model_vars = [
+        f"{var}: {ds[var].attrs['long_name']}" if "long_name" in ds[var].attrs else var
+        for var in x_keys
+    ]
+
+    text = "## Carbon tracker model \n \n ### Model variables: \n"
+
+    for item in model_vars:
+        text = text + f"- {item} \n"
+
+    for group in range(groups):
+        target_ds, prediction = groupwise_cross_validation(
+            df_group, group, x_keys, y_key
+        )
+        rmse = calculate_rmse(prediction["prediction_label"], target_ds[y_key])
+        rmse.to_netcdf(output_dir / f"rmse{group}.nc")
+        create_rmseplot(rmse)
+        plt.savefig(output_dir / f"rmseplot{group}.png")
+        plt.close()
+        create_scatterplot(prediction["prediction_label"], target_ds[y_key])
+        plt.savefig(output_dir / f"scatter{group}.png")
+        plt.close()
+        text = (
+            text
+            + f"\n ### Validation plots for group {group} \n"
```

Comment:
```suggestion
            + f"\n### Validation plots for group {group} \n"
```
unnecessary whitespace